### PR TITLE
Added new docs about the GameServer API

### LINKED ISSUE #161: GameServer API - docs
We should write some documentation for the GameServer API.

### PR DESCRIPTION
**Pull request by javier-op**
Resolves #161

### CODE REVIEW

File: docs/gameserverapi/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+title: GameServer API
+nav_order: 9
+has_children: true
+---
+
+# GameServer API
+
+The GameServer API is a RESTful API service that facilitates access to Thundernetes' Custom Resources: GameServerBuilds, GameServer, and GameServerDetails. It is an alternative for people who don't want to use tools like [kubectl](https://kubernetes.io/docs/reference/kubectl/kubectl/), and an easy way to integrate Thundernetes to your own applications. It also allows you to use our [Thundernetes UI](../thundernetesui/README.md).
+
+## How to install the GameServer API
```
Comment (dgkanatsios) [inline docs/gameserverapi/README.md:12]: should we add the link on how to secure the GameServer API?

File: docs/gameserverapi/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+layout: default
+title: GameServer API
+nav_order: 9
+has_children: true
+---
+
+# GameServer API
+
+The GameServer API is a RESTful API service that facilitates access to Thundernetes' Custom Resources: GameServerBuilds, GameServer, and GameServerDetails. It is an alternative for people who don't want to use tools like [kubectl](https://kubernetes.io/docs/reference/kubectl/kubectl/), and an easy way to integrate Thundernetes to your own applications. It also allows you to use our [Thundernetes UI](../thundernetesui/README.md).
```
Comment (dgkanatsios) [inline docs/gameserverapi/README.md:10]: GameServers maybe?

File: docs/gameserverapi/apidocs.md [hunk 3]
```diff
@@ -0,0 +1,775 @@
+---
```
Comment (dgkanatsios) [inline docs/gameserverapi/apidocs.md:1]: nice! was this generated? Can we provide a Postman collection?

Comment (javier-op) [reply docs/gameserverapi/apidocs.md:1]: No, I formatted this on my own haha

File: docs/gameserverapi/README.md (hunk 1 above)
Comment (javier-op) [reply docs/gameserverapi/README.md:12]: I think I do below, in how to install

File: docs/gameserverapi/apidocs.md (hunk 3 above)
Comment (dgkanatsios) [reply docs/gameserverapi/apidocs.md:1]: ouch - we should automate this next time, related to #160 I guess

File: docs/gameserverapi/README.md (hunk 1 above)
Comment (dgkanatsios) [reply docs/gameserverapi/README.md:12]: oh sorry, it was in gray colors because of the "_note" so I skipped it.

**Review by dgkanatsios — APPROVED** [review]

**Review by KalonJ-MSFT — APPROVED** [review]

**Pull request merged**